sesame/middleware.py

```diff
@@ -11,15 +15,23 @@ class AuthenticationMiddleware:
 
     def __init__(self, get_response):
         self.get_response = get_response
+        self._redirect_url = None
 
     def __call__(self, request):
+        self._redirect_url = None
         self.process_request(request)
-        return self.get_response(request)
+        response = self.get_response(request)
+
+        if self._redirect_url:
+            redirect_response = redirect(self._redirect_url)
+            redirect_response.content = response.content
```

Comment:
The tests rely on the authenticated user's username getting set in the response content. Does anything else from the original response need to be copied over?

Reply:
In fact the logic should be different here.

- If we want to redirect and remove the token, `__call__` shouldn't call `self.get_response(request)`; it should merely `return redirect(...)`. The redirect response doesn't need any content.
- In other cases, we proceed with `return self.get_response(request)`.